version: '3'

services:
  portainer:
    image: portainer/portainer-ce:2.20.2-alpine
    container_name: portainer
    restart: unless-stopped
    healthcheck:
      # https://github.com/portainer/portainer/issues/3572
      test: wget --no-verbose --tries=1 --spider http://localhost:9000 || exit 1
      interval: 15s
      timeout: 5s
      retries: 3
      start_period: 30s
    networks:
      - public
    volumes:
      - portainer-data:/data
      - /etc/localtime:/etc/localtime:ro
      - /var/run/docker.sock:/var/run/docker.sock
    labels:
      traefik.enable: 'true'
      traefik.http.routers.portainer.tls.certresolver: letsencrypt
      traefik.http.routers.portainer.rule: Host(`portainer.${BASE_HOST}`)
      traefik.http.routers.portainer.middlewares: sso-proxy@file
      traefik.http.services.portainer.loadbalancer.server.port: 9000
      flame.type: application
      flame.name: Portainer
      flame.url: https://portainer.${BASE_HOST}/#!/1/docker/containers
      flame.icon: console

networks:
  public: {}

volumes:
  portainer-data: {}
